### PR TITLE
Issue3130

### LINKED ISSUE #3130: Dataset merge when y's dimension is 1
### To reproduce,

```
a = dc.data.NumpyDataset(X=[1, 2, 3], y=[1, 2, 3])
b = dc.data.NumpyDataset(X=[1, 2, 3], y=[1, 2, 3])
dc.data.NumpyDataset.merge([a, b])

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Input In [39], in <cell line: 3>()
      1 a = dc.data.NumpyDataset(X=[1, 2, 3], y=[1, 2, 3])
      2 b = dc.data.NumpyDataset(X=[1, 2, 3], y=[1, 2, 3])
----> 3 dc.data.NumpyDataset.merge([a, b])

File ~/dc/deepchem/data/datasets.py:1093, in NumpyDataset.merge(datasets)
   1087   w = np.concatenate([w, dataset.w], axis=0)
   1088   ids = np.concatenate(
   1089       [ids, dataset.ids],
   1090       axis=0,
   1091   )
-> 1093 return NumpyDataset(X, y, w, ids, n_tasks=y.shape[1])

IndexError: tuple index out of range
```

The shape handling should be taken care internally when the numpy dataset is created.

### PR DESCRIPTION
**Pull request by advikavs**
# Pull Request Template

## Description

Fix #3130 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/data/datasets.py [hunk 1]
```diff
@@ -795,6 +795,8 @@ def __init__(self,
         w = np.ones((y.shape[0], 1), np.float32)
     if not isinstance(w, np.ndarray):
       w = np.array(w)
+    if y.shape == 1:
+      print("true")
```
Comment (rbharath) [inline deepchem/data/datasets.py:799]: There shouldn't be a print statement here

File: deepchem/data/datasets.py [hunk 2]
```diff
@@ -1089,7 +1091,9 @@ def merge(datasets: Sequence[Dataset]) -> "NumpyDataset":
           [ids, dataset.ids],
           axis=0,
       )
-
+    k = y.shape[0]
```
Comment (rbharath) [inline deepchem/data/datasets.py:1094]: Can you add a test for this new case? We should have a unit test that would have caught this error

File: deepchem/data/tests/test_datasets.py [hunk 3]
```diff
@@ -680,7 +680,13 @@ def test_merge():
     datasets.append(dataseti)
 
   new_data = dc.data.datasets.DiskDataset.merge(datasets)
-
+  x=[1,2, 3]
```
Comment (rbharath) [inline deepchem/data/tests/test_datasets.py:683]: Needs formatting fixes

**Comment by advikavs** [comment]
Reopened on new branch due to merge conflict.

**Pull request closed without merging**